react/Components/inputButtonArea.tsx

```diff
@@ -15,39 +16,90 @@ const InputButtonArea: FC = () => {
+  function submitFunctionValueButton(e: { preventDefault: () => void }) {
     e.preventDefault()
     updateGiftCardFunction()
   }
 
   function submitFunctionCodeButton(e: React.FormEvent<HTMLFormElement>) {
     e.preventDefault()
-    if (!navigator.clipboard) return
-    navigator.clipboard.writeText(code)
+    copyCode()
+  }
+
+  const currencyConfig = {
+    locale: 'pt-BR',
+    formats: {
+      number: {
+        BRL: {
+          style: 'currency',
+          currency: 'BRL',
+          minimumFractionDigits: 2,
+          maximumFractionDigits: 2,
+        },
+      },
+    },
+  }
+
+  const handleChange = (
+    event: { preventDefault: () => void },
+    value: string
+  ) => {
+    event.preventDefault()
+    setAddValueGiftCard(value)
   }
 
   return (
     <>
-      <div className="mb3 width-input mt5 inputs t-body mw9">
+      <div className="mb3 width-input mt5 inputs t-body mw9 rescue">
         {intl.formatMessage(input.valueLabel)}
-        <form onSubmit={(e) => submitFunctionValueButton(e)}>
-          <InputButton
-            placeholder={intl.formatMessage(input.valuePlaceholder)}
-            size="regular"
-            button={intl.formatMessage(input.valueButton)}
-            onChange={(e: { target: { value: string } }) =>
-              setAddValueGiftCard(e.target.value)
-            }
-            value={addValueGiftCard}
-            testId="input-button-test"
-          />
+        <div className="w-100">
+          <div className="desktop-or-mobile rescue">
+            <div className="lh-copy w-100">
+              <label className="vtex-input w-100">
+                <div
+                  className="vtex-input-prefix__group flex flex-row items-stretch overflow-hidden br2 bw1 b--solid b--muted-4 hover-b--muted-3 h-large "
+                  id="div-border-color"
+                >
+                  <IntlCurrencyInput
+                    defaultValue={addValueGiftCard}
+                    currency="BRL"
+                    config={currencyConfig}
+                    onFocus={(e: { preventDefault: () => void }) => {
+                      e.preventDefault()
+                      setValidation('')
+                      document
+                        .getElementById('div-border-color')
+                        ?.classList.remove('border-red')
+                    }}
+                    onBlur={handleChange}
+                    class="vtex-styleguide-9-x-input ma0 border-box vtex-styleguide-9-x-hideDecorators vtex-styleguide-9-x-noAppearance br2 bl-0 br--right   w-100 bn outline-0 bg-base c-on-base b--muted-4 hover-b--muted-3 t-body pr5 "
+                    disabled={loading}
+                  />
+                </div>
+              </label>
+            </div>
+            <div className="ml2 mt2">
+              <Button
+                onClick={(e: { preventDefault: () => void }) =>
+                  submitFunctionValueButton(e)
+                }
```

Comment:
The same ideia here, using `React.MouseEvent<HTMLButtonElement>` maybe a better way. Could you try it to check if works fine?
```suggestion
                onClick={submitFunctionValueButton}
```

Reply:
It's works fine
Thanks for the suggestion :)